lib/dai-plugin-mcd/src/ManagedCdp.js

```diff
@@ -32,6 +30,78 @@ export default class ManagedCdp {
     return art.times(rate).shiftedBy(-27);
   }
 
+  async getCollateralizationRatio() {
+    const [collateral, debt, price] = await Promise.all([
+      this.getCollateralValue(),
+      this.getDebtValue(),
+      this._cdpType.getPrice()
+    ]);
+
+    return (
+      collateral
+        .times(price)
+        .div(debt)
+        .toNumber() * 100
+    ).toFixed(2);
+  }
+
+  async getLiquidationPrice() {
+    const [liquidationRatio, debt, collateral] = await Promise.all([
+      this._cdpType.getLiquidationRatio(),
+      this.getDebtValue(),
+      this.getCollateralValue()
+    ]);
+    return debt
+      .times(liquidationRatio)
+      .div(collateral)
+      .toNumber()
+      .toFixed(2);
+  }
+
+  async isSafe() {
+    const [liqPrice, ethPrice] = await Promise.all([
+      this.getLiquidationPrice(),
+      this._cdpType.getPrice()
+    ]);
+    return ethPrice.gte(liqPrice);
+  }
+
+  async getLockedCollateral() {
+    const [debt, liquidationRatio, price, collateral] = await Promise.all([
+      this.getDebtValue(),
+      this._cdpType.getLiquidationRatio(),
+      this._cdpType.getPrice(),
+      this.getCollateralValue()
+    ]);
+    const lockedCollateral =
```

Comment:
Hm, this name is a little confusing to me.  Maybe `minCollateralRequired`?

Reply:
can you say more here?

Reply:
we already use the term "locked" at maker to refer to all the collateral in the CDP, not just the minimum safe amount

Reply:
nvm I understand